src/test/java/org/zalando/nakadi/service/subscription/StreamingContextTest.java

```diff
@@ -145,9 +145,13 @@ public void onEnter() {
             }
         });
         t.start();
-        t.join(2000);
+        t.join(1000);
+
+        t = new Thread(() -> ctxSpy.onNodeShutdown());
+        t.start();
+        t.join(1000);
 
-        ctxSpy.onNodeShutdown();
+        Thread.sleep(1000);
```

Comment:
onNodeShutdown switches the state to CleanupState which is processed by the first thread in the test, but after that CleanupState itself changes the state to DummyState which also has be processed by the first thread. Probably, I do not need the join on the second thread, because Thread.sleep should be enough.